contrib/kafka-humio-gateway/main.go

```diff
@@ -162,15 +166,19 @@ func main() {
 
 // Consumer represents a Sarama consumer group consumer
 type Consumer struct {
-	ready chan bool
+	setupOnce  bool
+	readyWg    *sync.WaitGroup
 	httpClient http.Client
-	config TransportConfig
+	config     TransportConfig
 }
 
 // Setup is run at the beginning of a new session, before ConsumeClaim
 func (consumer *Consumer) Setup(sarama.ConsumerGroupSession) error {
-	// Mark the consumer as ready
-	close(consumer.ready)
+	if !consumer.setupOnce {
+		// Allow the main thread to proceed
+		consumer.readyWg.Done()
+	}
+	consumer.setupOnce = true
```

Comment:
All of this can be replaced with an https://pkg.go.dev/sync#Once

Reply:
Yep, that would be nicer.